### DISCUSSION TITLE
File system or Azure blob cache

### DISCUSSION THREAD

**Discussion by olaj** [SixLabors/ImageSharp.Web#153] [Q&A]
I sometimes noticed spike in latency (10-30 sec) and in between it looks like it average on 50-100ms on my ImageSharp deployment on a Azure App Service.

I use the Azure Blob cache atm and I'm a bit curious if that could be the issue. I realize i need to research this more but i ran into this post https://github.com/SixLabors/ImageSharp.Web/issues/140 and it got me thinking that i maybe should use local file system cache instead.

What are the major pros and cons?

Can i use both? I mean a layer for file system cache and if it missed it goes to the azure cache?

As a side note i do see a few 500 errors.

**Comment by JimBobSquarePants** [comment]
I wouldn’t consider 500 errors a side note! They’re critical failures. 

We’ve published a release v1.0.2 since that issue and fix. Are you using the latest? 

https://github.com/SixLabors/ImageSharp.Web/releases/tag/v1.0.2

**Reply by olaj** [reply]
Ok! Thanks :) I will look into the 500 errors and do some log digging.

Will also try the new version and see if i notice any changes. But still, what are the pros and cons of a filesystem cache vs using a blob storage, what would be consider the "fastest"? Maybe it's not possible to anwer without doing some test.

**Reply by JimBobSquarePants** [reply]
Well... They're used for different things. 

If you only ever have a single application reading and writing to the cache then use the filesystem cache. It's faster because there are no Http Requests involved to Azure. 

If you are horizontally scaling your application and will have multiple instances sharing a cache you should use the Azure Blob cache since that allows concurrent shared access.
